rs/bitcoin/kyt/src/fetch.rs

```diff
@@ -218,8 +218,16 @@ pub trait FetchEnv {
                         futures.push(do_fetch)
                     }
                     Fetched(fetched) => {
-                        let address = &fetched.tx.outputs[input.vout as usize];
-                        state::set_fetched_address(txid, index, address.clone());
+                        if let Some(address) = &fetched.tx.outputs[input.vout as usize] {
+                            state::set_fetched_address(txid, index, address.clone());
+                        } else {
+                            // This error shouldn't happen unless blockdata is corrupted.
+                            return CheckTransactionRetriable::TransientInternalError(format!(
```

Comment:
Why is this considered a transient error?

Reply:
After a second thought, I think you are right. This should be a irrecoverable error for 2 reasons:
1. The next call to `check_transaction` will return the same error, because the dependency transaction is already fetched into cache, and will not be retried.
2. The transaction data already passed txid check, so most likely all RPC providers would return the same data, and the error will not go away by trying with another provider.
WDYT?

Reply:
> The transaction data already passed txid check,

To me this seems like the key argument why this error should indeed be irrecoverable.